### ISSUE TITLE
Testing: reporting

### ISSUE THREAD

**Issue by Koenkk** [Koenkk/zigbee2mqtt#1064]
Reporting is a new feature which is currently only available in the dev branch. When reporting is enabled, the device will report their state changes to zigbee2mqtt. This is especially handy when the state of a bulb is changed by e.g. a remote or via groups.

To enable report, add the following to your `configuration.yaml`:

```yaml
advanced:
  report: true
```

As reporting is a feature of the device, not all devices may support this. 

**Current findings**
- IKEA TRADFRI bulbs work as expected
- Hue bulbs only report on/off state (limitation of Hue bulbs) (https://github.com/Koenkk/zigbee2mqtt/issues/1064#issuecomment-468962505)

**Comment by milakov** [comment]
Hi!

Gledopto GL-B-007Z:

Changing color temperature triggers this (along with actual ct of the bulb changing):
```
zigbee2mqtt:error 2019-2-10 22:06:26 No converter available for 'color_temp' (166)
```

Changing brightness doesn't trigger anything in the logs (but the brightness of the physical bulb actually changes, HA still thinks the brightness is unchanged).

**Comment by Koenkk** [comment]
@milakov should be fixed in the latest dev branch, if not, please open a separate issue as this is not related to this issue.

**Comment by sylarevan** [comment]
Just tried with a group of two Hue bulbs. Changing the state (ON or OFF) of the group does not report the individual change of state of each bulb.
Logs in debug mode are [here](https://pastebin.com/JBQQmyU9).

**Comment by JumpmanJunior** [comment]
I tried to activate reporting for a Philips and a Ikea bulb. But when I control the bulbs with a bound Philips dimmer switch the status changes are not reported. Here is the log for the Ikea bulb:


> 2019-2-10 22:58:08 - debug: Setup reporting for 0x000d6ffffe1df5f5 - genOnOff - onOff
> 2019-2-10 22:58:09 - debug: Setup reporting for 0x000d6ffffe1df5f5 - genLevelCtrl - currentLevel
> 2019-2-10 22:58:09 - debug: Successfully setup reporting for 0x000d6ffffe1df5f5 - genOnOff - onOff
> 2019-2-10 22:58:09 - debug: Successfully setup reporting for 0x000d6ffffe1df5f5 - genLevelCtrl - currentLevel
> 2019-2-10 22:58:09 - debug: Setup reporting for 0x000d6ffffe1df5f5 - lightingColorCtrl - currentX
> 2019-2-10 22:58:09 - debug: Successfully setup reporting for 0x000d6ffffe1df5f5 - lightingColorCtrl - currentX
> 2019-2-10 22:58:09 - debug: Setup reporting for 0x000d6ffffe1df5f5 - lightingColorCtrl - currentY
> 2019-2-10 22:58:09 - debug: Successfully setup reporting for 0x000d6ffffe1df5f5 - lightingColorCtrl - currentY

**Comment by way2many** [comment]
I tested reporting of two Tradfri GU10 in my hallway this morning. I switched them off with the wire switch (220V) and put them back on. This is the log:

> 2/12/2019, 6:19:55 AM - debug: Saving state to file /app/data/state.json
2/12/2019, 6:21:53 AM - debug: Received zigbee message of type 'endDeviceAnnce' with data '"0x90fd9ffffe1a7413"' of device 'TRADFRI bulb GU10 W 400lm' (0x90fd9ffffe1a7413)
2/12/2019, 6:21:53 AM - debug: Setup reporting for 0x90fd9ffffe1a7413 - genOnOff - onOff
2/12/2019, 6:21:53 AM - debug: Received zigbee message of type 'endDeviceAnnce' with data '"0x90fd9ffffe2a3f57"' of device 'TRADFRI bulb GU10 W 400lm' (0x90fd9ffffe2a3f57)
2/12/2019, 6:21:54 AM - debug: Setup reporting for 0x90fd9ffffe1a7413 - genLevelCtrl - currentLevel
2/12/2019, 6:21:54 AM - debug: Successfully setup reporting for 0x90fd9ffffe1a7413 - genOnOff - onOff
2/12/2019, 6:21:54 AM - debug: Successfully setup reporting for 0x90fd9ffffe1a7413 - genLevelCtrl - currentLevel
2/12/2019, 6:21:54 AM - debug: Setup reporting for 0x90fd9ffffe2a3f57 - genOnOff - onOff
2/12/2019, 6:21:54 AM - debug: Successfully setup reporting for 0x90fd9ffffe2a3f57 - genOnOff - onOff
2/12/2019, 6:21:54 AM - debug: Setup reporting for 0x90fd9ffffe2a3f57 - genLevelCtrl - currentLevel
2/12/2019, 6:21:54 AM - debug: Successfully setup reporting for 0x90fd9ffffe2a3f57 - genLevelCtrl - currentLevel
2/12/2019, 6:21:55 AM - debug: Received zigbee message of type 'devInterview' with data '"0x90fd9ffffe1a7413"'
2/12/2019, 6:21:56 AM - debug: Received zigbee message of type 'devInterview' with data '"0x90fd9ffffe1a7413"'
2/12/2019, 6:21:56 AM - debug: Received zigbee message of type 'devInterview' with data '"0x90fd9ffffe1a7413"'
2/12/2019, 6:21:57 AM - debug: Received zigbee message of type 'devInterview' with data '"0x90fd9ffffe1a7413"'
2/12/2019, 6:21:57 AM - debug: Received zigbee message of type 'devInterview' with data '"0x90fd9ffffe1a7413"'
2/12/2019, 6:21:58 AM - debug: Received zigbee message of type 'devInterview' with data '"0x90fd9ffffe1a7413"'
2/12/2019, 6:21:58 AM - debug: Received zigbee message of type 'devInterview' with data '"0x90fd9ffffe1a7413"'
2/12/2019, 6:21:58 AM - debug: Received zigbee message of type 'devInterview' with data '"0x90fd9ffffe1a7413"'
2/12/2019, 6:21:59 AM - debug: Received zigbee message of type 'devInterview' with data '"0x90fd9ffffe1a7413"'
2/12/2019, 6:21:59 AM - debug: Received zigbee message of type 'devInterview' with data '"0x90fd9ffffe1a7413"'
2/12/2019, 6:21:59 AM - debug: Received zigbee message of type 'devIncoming' with data '"0x90fd9ffffe1a7413"' of device 'TRADFRI bulb GU10 W 400lm' (0x90fd9ffffe1a7413)
2/12/2019, 6:21:59 AM - info: Device incoming...
2/12/2019, 6:21:59 AM - info: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"type":"pairing","message":"device incoming"}'
2/12/2019, 6:21:59 AM - debug: Received zigbee message of type 'devStatus' with data '"online"' of device 'TRADFRI bulb GU10 W 400lm' (0x90fd9ffffe1a7413)
2/12/2019, 6:22:01 AM - debug: Received zigbee message of type 'devInterview' with data '"0x90fd9ffffe2a3f57"'
2/12/2019, 6:22:01 AM - debug: Received zigbee message of type 'devInterview' with data '"0x90fd9ffffe2a3f57"'
2/12/2019, 6:22:01 AM - debug: Received zigbee message of type 'devInterview' with data '"0x90fd9ffffe2a3f57"'
2/12/2019, 6:22:02 AM - debug: Received zigbee message of type 'devInterview' with data '"0x90fd9ffffe2a3f57"'
2/12/2019, 6:22:02 AM - debug: Received zigbee message of type 'devInterview' with data '"0x90fd9ffffe2a3f57"'
2/12/2019, 6:22:03 AM - debug: Received zigbee message of type 'devInterview' with data '"0x90fd9ffffe2a3f57"'
2/12/2019, 6:22:03 AM - debug: Received zigbee message of type 'devInterview' with data '"0x90fd9ffffe2a3f57"'
2/12/2019, 6:22:03 AM - debug: Received zigbee message of type 'devInterview' with data '"0x90fd9ffffe2a3f57"'
2/12/2019, 6:22:04 AM - debug: Received zigbee message of type 'devInterview' with data '"0x90fd9ffffe2a3f57"'
2/12/2019, 6:22:04 AM - debug: Received zigbee message of type 'devInterview' with data '"0x90fd9ffffe2a3f57"'
2/12/2019, 6:22:04 AM - debug: Received zigbee message of type 'devIncoming' with data '"0x90fd9ffffe2a3f57"' of device 'TRADFRI bulb GU10 W 400lm' (0x90fd9ffffe2a3f57)
2/12/2019, 6:22:04 AM - info: Device incoming...
2/12/2019, 6:22:04 AM - info: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"type":"pairing","message":"device incoming"}'
2/12/2019, 6:22:04 AM - debug: Received zigbee message of type 'devStatus' with data '"online"' of device 'TRADFRI bulb GU10 W 400lm' (0x90fd9ffffe2a3f57)
2/12/2019, 6:24:55 AM - debug: Saving state to file /app/data/state.json

I have now multiple questions:

- What is "endDeviceAnnce" and why did the bulb publish this?
- Am I correct to interpreting this as when physically switched off, the bulb lose it's reporting function (if it has one)?
- Am I also correct that when physically switched off, the bulb re-pair itself ?

As I'm unable to bind my Hue Dimmer Switch to my Hue suspension Fair I cannot test the reporting function on those one.

**Comment by nanosonde** [comment]
I am also trying to achieve the same.
I have a TRADFRI remote and TRADFRI driver paired to zigbee2mqtt. I have found out the group ID of the remote by sniffing. Then I have created a group with this ID and added the TRADFRI driver to this group. I can successfully control the driver with the remote and z2m. 

However, I do not get any reports in z2m if it is controlled by the remote.

**Comment by nanosonde** [comment]
I am not seeing any attribue reports with the sniffer either.

**Comment by nanosonde** [comment]
From the ZCL Configure Reporting Message I see that there is a min interval of 1 and a max interval of 3600. 3600s would be 1h.
So how does the device choose which to use? Is it possible to lower the value to 10s and see what happens?

**Comment by Koenkk** [comment]
@nanosonde just change https://github.com/Koenkk/zigbee2mqtt/blob/dev/lib/extension/reporting.js#L12 to `10` .

**Comment by nanosonde** [comment]
@Koenkk I have changed it to 10s and verified with the sniffer that this is sent over the wire.
Unfortunately, it does not work too.

**Comment by nanosonde** [comment]
@Koenkk Somewhere in the deconz-plugin issues I have read that reporting for a TRADFRI device(?) only worked when it is read back. I am not absolutely sure anymore.

**Comment by nanosonde** [comment]
Got it working. I have disabled the generic reporting setup using "report: true".
Instead I have changed device.js for the 10W dimmer similiar to the TRADFRI power outlet.

```
    {
        zigbeeModel: ['TRADFRI transformer 10W', 'TRADFRI Driver 10W'],
        model: 'ICPSHC24-10EU-IL-1',
        vendor: 'IKEA',
        description: 'TRADFRI driver for wireless control (10 watt)',
        //extend: generic.light_onoff_brightness,
        fromZigbee: [fz.ignore_onoff_change, fz.state],
        toZigbee: [tz.on_off],
        configure: (ieeeAddr, shepherd, coordinator, callback) => {
            const device = shepherd.find(ieeeAddr, 1);
            const cfg = {direction: 0, attrId: 0, dataType: 16, minRepIntval: 0, maxRepIntval: 1000, repChange: 0};
            const actions = [
                (cb) => device.bind('genOnOff', coordinator, cb),
                (cb) => device.foundation('genOnOff', 'configReport', [cfg], foundationCfg, cb),
            ];

            execute(device, actions, callback);
        },
    },
```

This way the driver immediately reports state changes.

**Comment by nanosonde** [comment]
@Koenkk 
Maybe I missing something, but is the binding to the coordinator missing if reports are enabled using the generic mechansim? Or is this only required for TRADFRI devices?

**Comment by nanosonde** [comment]
BTW: It looks like this with the above code change:

```
  zigbee2mqtt:debug 2/13/2019, 7:10:02 PM Received zigbee message of type 'cmdToggle' with data '{"cid":"genOnOff","data":{}}' of device 'TRADFRI remote control' (0xd0cf5efffe0bxxxx)
  zigbee2mqtt:info 2/13/2019, 7:10:02 PM MQTT publish: topic 'zigbee2mqtt/ikea_remotecontrol', payload '{"action":"toggle","linkquality":10,"last_seen":"2019-02-13T19:10:02.922Z","elapsed":5628497}'
  zigbee2mqtt:debug 2/13/2019, 7:10:02 PM Received zigbee message of type 'attReport' with data '{"cid":"genOnOff","data":{"onOff":1}}' of device 'TRADFRI transformer 10W' (0x000d6ffffe2cyyyy)
  zigbee2mqtt:info 2/13/2019, 7:10:02 PM MQTT publish: topic 'zigbee2mqtt/ikea_driver', payload '{"state":"ON","last_seen":"2019-02-13T19:10:02.955Z","linkquality":7,"elapsed":628194}'
```

**Comment by Koenkk** [comment]
@nanosonde I've also added the binding, meaning it should be the same as your custom `configure` now. Can you factory reset/re-pair the device and verify that it works with `report: true`?

**Comment by nanosonde** [comment]
@Koenkk  I have tested latest dev and seems to be working now as with the custom configure.
As I did a complete ./update.sh I some error now concerning the remote control:
```
  zigbee2mqtt:debug 2/13/2019, 8:42:05 PM Coordinator is already in group '611'
  zigbee2mqtt:error 2/13/2019, 8:42:05 PM Failed to apply coordinator group for IKEA TRADFRI remote control (0xd0cf5efffe0bbf60)
  zigbee2mqtt:debug 2/13/2019, 8:42:05 PM Coordinator is already in group '99'
```
Did you change anything here?
This was working before.

If I remove "coordinator_group" from the device config then this error is gone and I also receive the reports from button pushes.

**Comment by nanosonde** [comment]
> 
> 
> @Koenkk I have tested latest dev and seems to be working now as with the custom configure.
> As I did a complete ./update.sh I experience an error now concerning the remote control:
> 
> ```
>   zigbee2mqtt:debug 2/13/2019, 8:42:05 PM Coordinator is already in group '611'
>   zigbee2mqtt:error 2/13/2019, 8:42:05 PM Failed to apply coordinator group for IKEA TRADFRI remote control (0xd0cf5efffe0bxxxx)
>   zigbee2mqtt:debug 2/13/2019, 8:42:05 PM Coordinator is already in group '99'
> ```
Ok, I will answer that by myself after studying the code.
So actually this is NOT a failure, but it is just as the previous message tells it: the ZDO of the coordinator already has the group added. So this is a bit misleading.

**Comment by Koenkk** [comment]
@nanosonde you are right, fixed.

**Comment by Koenkk** [comment]
Groups will now also report their state (on `zigbee2mqtt/[GROUP_FRIENDLY_NAME`). (dev branch)

**Comment by dreimer1986** [comment]
```
  zigbee2mqtt:error 2019-2-15 10:15:42 Failed to setup reporting for 0x84182600000b8298 - lightingColorCtrl - currentX - (Error: AF data request fails, status code: 233. MAC no ack.)
  zigbee2mqtt:error 2019-2-15 10:15:44 Failed to setup reporting for 0x84182600000b79b7 - genLevelCtrl - currentLevel - (Error: AF data request fails, status code: 233. MAC no ack.)
  zigbee2mqtt:warn 2019-2-15 10:15:45 Failed to configure Motion (0x000d6f000e059aae) ('Error: AF data request fails, status code: 240. MAC transaction expired.') (attempt #1)
  zigbee2mqtt:warn 2019-2-15 10:15:45 This can be ignored if the device is working properly
  zigbee2mqtt:error 2019-2-15 10:15:45 Failed to setup reporting for 0x7cb03eaa00a1184a - lightingColorCtrl - currentX - (Error: AF data request fails, status code: 233. MAC no ack.)
  zigbee2mqtt:error 2019-2-15 10:15:46 Failed to setup reporting for 0x84182600000b743a - genOnOff - onOff - (Error: AF data request fails, status code: 205. No network route. Please confirm that the device has (re)joined the network.)
  zigbee2mqtt:error 2019-2-15 10:15:47 Failed to setup reporting for 0x84182600000b743a - lightingColorCtrl - colorTemperature - (Error: AF data request fails, status code: 205. No network route. Please confirm that the device has (re)joined the network.)
  zigbee2mqtt:error 2019-2-15 10:15:47 Failed to setup reporting for 0x84182600000b743a - genLevelCtrl - currentLevel - (Error: AF data request fails, status code: 205. No network route. Please confirm that the device has (re)joined the network.)
  zigbee2mqtt:error 2019-2-15 10:15:47 Failed to setup reporting for 0x84182600000b83ef - genOnOff - onOff - (Error: AF data request fails, status code: 233. MAC no ack.)
  zigbee2mqtt:error 2019-2-15 10:15:47 Failed to setup reporting for 0x84182600000b83ef - genLevelCtrl - currentLevel - (Error: AF data request fails, status code: 233. MAC no ack.)
  zigbee2mqtt:error 2019-2-15 10:15:47 Failed to setup reporting for 0x84182600000b743a - lightingColorCtrl - currentY - (Error: AF data request fails, status code: 205. No network route. Please confirm that the device has (re)joined the network.)
  zigbee2mqtt:error 2019-2-15 10:15:47 Failed to setup reporting for 0x84182600000b743a - lightingColorCtrl - currentX - (Error: AF data request fails, status code: 205. No network route. Please confirm that the device has (re)joined the network.)
  zigbee2mqtt:error 2019-2-15 10:15:48 Failed to setup reporting for 0x8418260000ca9270 - lightingColorCtrl - currentY - (Error: AF data request fails, status code: 233. MAC no ack.)
  zigbee2mqtt:error 2019-2-15 10:15:50 Failed to setup reporting for 0x7cb03eaa00a1184a - genLevelCtrl - currentLevel - (Error: AF data request fails, status code: 205. No network route. Please confirm that the device has (re)joined the network.)
```
I added reporting to all my bulbs, in a group or not. Reporting does not work, but on startup you see this after it reconnected to all Zigbee devices. Do I have to repair all my bulbs to make it work?

EDIT: Debug loglevel: https://h.kremowka.xyz/etayakilox.coffeescript

**Comment by Koenkk** [comment]
@dreimer1986 can you control the bulb individually (turn on/off)?

**Comment by alinelena** [comment]
@Koenkk i get he same error as @dreimer1986 

```
  2/16/2019, 8:21:25 AM - debug: Setup binding for 0x000b57fffeb0c77e - genOnOff
  2/16/2019, 8:21:25 AM - debug: Setup reporting for 0x000b57fffeb0c77e - genOnOff - onOff
  2/16/2019, 8:21:25 AM - debug: Setup binding for 0x000b57fffeb0c77e - genLevelCtrl
  2/16/2019, 8:21:25 AM - debug: Setup reporting for 0x000b57fffeb0c77e - genLevelCtrl - currentLevel
)>2/16/2019, 8:21:25 AM - error: Failed to setup reporting for 0x000b57fffeb0c77e - genOnOff - onOff - (Error: AF data request fails, status code: 205. No network route. Please confirm that the device has (re)joined the network.)
  2/16/2019, 8:21:25 AM - debug: Setup binding for 0x000b57fffeb24cb3 - genOnOff
  2/16/2019, 8:21:26 AM - debug: Setup reporting for 0x000b57fffeb24cb3 - genOnOff - onOff
  2/16/2019, 8:21:26 AM - debug: Setup binding for 0x000b57fffeb24cb3 - genLevelCtrl
  2/16/2019, 8:21:26 AM - error: Failed to setup reporting for 0x000b57fffeb0c77e - genLevelCtrl - currentLevel - (Error: AF data request fails, status code: 205. No network route. Please confirm that the device has (re)joined the network
```
much later I get a binding success message for both bulbs here.
nothing more about reporting except the error. However reporting works and I can control them individually too or via the group.

I am on this
```
  2/16/2019, 8:21:21 AM - info: Starting zigbee2mqtt version 1.1.1 (commit #577bb0d)
```

I think that somehow one sets reporting too early

**Comment by dreimer1986** [comment]
@Koenkk yes, controlling any of the bulbs works fine. Only reporting is not and the log fills a bit more when activated.

**Comment by alinelena** [comment]
@dreimer1986  this is what is in my log
```
2/16/2019, 4:01:38 PM - info: MQTT publish: topic 'zigbee2mqtt/bedroom_lights', payload '{"state":"OFF","brightness":255}'
2/16/2019, 4:01:38 PM - debug: Received zigbee message of type 'attReport' with data '{"cid":"genOnOff","data":{"onOff":0}}' of device 'TRADFRI bulb E26 W opal 1000lm' (0x000b57fffeb0c77e)
2/16/2019, 4:01:38 PM - info: MQTT publish: topic 'zigbee2mqtt/ikea_bedroom_1', payload '{"state":"OFF","linkquality":49,"brightness":254}'
2/16/2019, 4:01:38 PM - debug: Received zigbee message of type 'devChange' with data '{"cid":"genOnOff","data":{"onOff":0}}' of device 'TRADFRI bulb E26 W opal 1000lm' (0x000b57fffeb0c77e)
2/16/2019, 4:01:38 PM - info: MQTT publish: topic 'zigbee2mqtt/ikea_bedroom_1', payload '{"state":"OFF","linkquality":49,"brightness":254}'
2/16/2019, 4:01:38 PM - debug: Received zigbee message of type 'attReport' with data '{"cid":"genOnOff","data":{"onOff":0}}' of device 'TRADFRI bulb E26 W opal 1000lm' (0x000b57fffeb24cb3)
2/16/2019, 4:01:38 PM - info: MQTT publish: topic 'zigbee2mqtt/ikea_bedroom_2', payload '{"state":"OFF","linkquality":15,"brightness":254}'
2/16/2019, 4:01:38 PM - debug: Received zigbee message of type 'devChange' with data '{"cid":"genOnOff","data":{"onOff":0}}' of device 'TRADFRI bulb E26 W opal 1000lm' (0x000b57fffeb24cb3)
2/16/2019, 4:01:38 PM - info: MQTT publish: topic 'zigbee2mqtt/ikea_bedroom_2', payload '{"state":"OFF","linkquality":15,"brightness":254}'  
```

annoying bit is the messages are reported twice

**Comment by dreimer1986** [comment]
Nope, no report over here. Will repair the devices later these days. Maybe this finally fixes that and maybe my Smart+ Motion sensor to not only send me temperature information, but MOTION too ^^

**Comment by Koenkk** [comment]
@jbmbn greatly helped in improving the reporting functionality.

This is not configured per device anymore, but a global setting now, see OP.

**Comment by BudBundi** [comment]
It seems that GL-S-007Z - Gledopto Smart RGBW GU10 doesn't support reporting, no report after change over remote or homeassistant.
Also I see this error in log: 
`zigbee2mqtt:info 2/27/2019, 5:59:09 PM Zigbee publish to device '0x00124b001b6ae5e4', genOnOff - read - [{"attrId":0}] - {"manufSpec":0,"disDefaultRsp":0} - null
zigbee2mqtt:error 2/27/2019, 5:59:39 PM Zigbee publish to device '0x00124b001b6ae5e4', genOnOff - read - [{"attrId":0}] - {"manufSpec":0,"disDefaultRsp":0} - null failed with error Error: Timed out after 30000 ms`

Same for GL-C-008 - Gledopto Zigbee LED controller RGB + CCT / RGBW / WWCW / Dimmer

**Comment by JumpmanJunior** [comment]
I have tried to setup reporting with the newest dev version. I am controlling a Hue bulb with a bound Hue dimmer switch. But the bulb only reports ons and offs but no brightness changes.

> 2019-3-1 21:21:57 - debug: Setup reporting for 0x00178801026cd5d3 - genOnOff - 1
> 2019-3-1 21:21:58 - debug: Setup reporting for 0x00178801026cd5d3 - genLevelCtrl - 1
> 2019-3-1 21:21:59 - debug: Successfully setup reporting for 0x00178801026cd5d3 - genOnOff - 1
> 2019-3-1 21:21:59 - debug: Successfully setup reporting for 0x00178801026cd5d3 - genLevelCtrl - 1
> 2019-3-1 21:56:49 - debug: Received zigbee message of type 'attReport' with data '{"cid":"genOnOff","data":{"onOff":0}}' of device 'LWB010' (0x00178801026cd5d3)
> 2019-3-1 21:56:49 - info: MQTT publish: topic 'zigbee2mqtt/xBulb', payload '{"state":"OFF","brightness":50,"linkquality":28}'
> 2019-3-1 21:56:49 - debug: Received zigbee message of type 'devChange' with data '{"cid":"genOnOff","data":{"onOff":0}}' of device 'LWB010' (0x00178801026cd5d3)
> 2019-3-1 21:56:49 - info: MQTT publish: topic 'zigbee2mqtt/xBulb', payload '{"state":"OFF","brightness":50,"linkquality":28}'
> 2019-3-1 21:56:56 - debug: Received zigbee message of type 'attReport' with data '{"cid":"genOnOff","data":{"onOff":1}}' of device 'LWB010' (0x00178801026cd5d3)
> 2019-3-1 21:56:56 - info: MQTT publish: topic 'zigbee2mqtt/xBulb', payload '{"state":"ON","brightness":50,"linkquality":21}'
> 2019-3-1 21:56:56 - debug: Received zigbee message of type 'devChange' with data '{"cid":"genOnOff","data":{"onOff":1}}' of device 'LWB010' (0x00178801026cd5d3)
> 2019-3-1 21:56:56 - info: MQTT publish: topic 'zigbee2mqtt/xBulb', payload '{"state":"ON","brightness":50,"linkquality":21}'

There are no logs if I change the brightness. The reported brightness of 50 seems to be a cached value and is not correct.

**Comment by jbmbn** [comment]
The reporting functionality is configured to only report on/off periodically. Other reports are generated only on change of the attribute.

One thing I think is still missing is retrieval of the current state of other attributes when a devices comes online (or zigbee2mqtt get restarted).

I've seen some articles that Philips Hue does not properly support the reporting function but I don't know what's true. If you want to try periodical reporting on the current level, you could try to change the value 3600 in deviceReport.js to 30 or so. Reports should be generated every 30 seconds. If that doesn't work, either the Philips Hue does not support reporting current level or it uses a different attribute that's not configured in deviceReport.js

**Comment by JumpmanJunior** [comment]
Indeed the problem seems to be caused by the Philips bulb. I have now tried the same with an Ikea bulb and the immediate reporting of brightness changes was working without problems.

I have also tested the mentioned periodical reporting. The Philips bulb sends no periodical reports at all. The Ikea bulb sends these as expected. So with the Philips bulb at the moment only immediate reporting of state changes (on/off) is working.

**Comment by BudBundi** [comment]
With the docker dev image from today and the latest dev firmware I get some strange warnings. I think its reporting related:
`zigbee2mqtt:warn 3/4/2019, 9:29:36 PM No converter available for 'GL-C-008' with cid 'genBasic', type 'attReport' and data '{"cid":"genBasic","data":{"manufacturerName":"GLEDOPTO","modelId":"GL-C-008","powerSource":1}}'
  zigbee2mqtt:warn 3/4/2019, 9:29:36 PM No converter available for 'GL-C-008' with cid 'genBasic', type 'devChange' and data '{"cid":"genBasic","data":{"manufacturerName":"GLEDOPTO","modelId":"GL-C-008","powerSource":1}}'
  zigbee2mqtt:warn 3/4/2019, 9:30:07 PM No converter available for 'LED1537R6' with cid 'genBasic', type 'attReport' and data '{"cid":"genBasic","data":{"manufacturerName":"IKEA of Sweden","modelId":"TRADFRI bulb GU10 WS 400lm","powerSource":1}}'
  zigbee2mqtt:warn 3/4/2019, 9:30:07 PM No converter available for 'LED1537R6' with cid 'genBasic', type 'attReport' and data '{"cid":"genBasic","data":{"zclVersion":1,"appVersion":17,"stackVersion":87,"hwVersion":1,"manufacturerName":"IKEA of Sweden"}}'
  zigbee2mqtt:warn 3/4/2019, 9:30:07 PM No converter available for 'LED1537R6' with cid 'genBasic', type 'attReport' and data '{"cid":"genBasic","data":{"10":{"type":"Buffer","data":[network_key]},"modelId":"TRADFRI bulb GU10 WS 400lm","dateCode":"20170331","powerSource":1}}'
  zigbee2mqtt:warn 3/4/2019, 9:30:07 PM No converter available for 'LED1537R6' with cid 'genBasic', type 'devChange' and data '{"cid":"genBasic","data":{"10":[network_key]}}'
  zigbee2mqtt:warn 3/4/2019, 9:30:08 PM No converter available for 'LED1537R6' with cid 'genBasic', type 'attReport' and data '{"cid":"genBasic","data":{"65533":1}}'
  zigbee2mqtt:warn 3/4/2019, 9:30:08 PM No converter available for 'LED1537R6' with cid 'genIdentify', type 'attReport' and data '{"cid":"genIdentify","data":{"65533":1,"identifyTime":0}}'
  zigbee2mqtt:warn 3/4/2019, 9:30:08 PM No converter available for 'LED1537R6' with cid 'genGroups', type 'attReport' and data '{"cid":"genGroups","data":{"65533":1,"nameSupport":0}}'
  zigbee2mqtt:warn 3/4/2019, 9:30:09 PM No converter available for 'LED1537R6' with cid 'genScenes', type 'attReport' and data '{"cid":"genScenes","data":{"count":0,"currentScene":255,"currentGroup":0,"sceneValid":0,"nameSupport":0}}'
  zigbee2mqtt:warn 3/4/2019, 9:30:09 PM No converter available for 'LED1537R6' with cid 'genScenes', type 'attReport' and data '{"cid":"genScenes","data":{"65533":1}}' 
  zigbee2mqtt:warn 3/4/2019, 9:30:11 PM No converter available for 'LED1537R6' with cid 'haDiagnostic', type 'attReport' and data '{"cid":"haDiagnostic","data":{"numberOfResets":0,"macRxBcast":0,"macTxBcast":0,"aPSRxBcast":0,"aPSTxBcast":0}}'
  zigbee2mqtt:warn 3/4/2019, 9:30:12 PM No converter available for 'LED1537R6' with cid 'haDiagnostic', type 'attReport' and data '{"cid":"haDiagnostic","data":{"nwkFcFailure":0,"packetBufferAllocateFailures":0,"averageMacRetryPerApsMessageSent":0,"lastMessageLqi":132,"lastMessageRssi":-67}}'
  zigbee2mqtt:warn 3/4/2019, 9:30:12 PM No converter available for 'LED1537R6' with cid 'haDiagnostic', type 'devChange' and data '{"cid":"haDiagnostic","data":{"lastMessageLqi":132,"lastMessageRssi":-67}}'
  zigbee2mqtt:warn 3/4/2019, 9:30:12 PM No converter available for 'LED1537R6' with cid 'haDiagnostic', type 'attReport' and data '{"cid":"haDiagnostic","data":{"65533":1}}'
  zigbee2mqtt:warn 3/4/2019, 9:30:12 PM No converter available for 'LED1537R6' with cid 'lightLink', type 'attReport' and data '{"cid":"lightLink","data":{"65533":1}}'`

With `zigbee2mqtt:warn 3/4/2019, 9:29:36 PM Please see: https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html.` in between

**Comment by alinelena** [comment]
@BudBundi here the same but on a normal install... also plenty of errors too related to reporting, one needs to turn on/off the device to repair. 
repairing new devices is plugged by error node descriptor stuff. 
firmware keeps hitting errors are restart 60s to repair... thought this was fixed.

**Comment by Koenkk** [comment]
@BudBundi fixed in the latest dev branch.

**Comment by milakov** [comment]
I am getting this with the very latest dev both for GL-B-007Z and GL-B-008Z:
```
  zigbee2mqtt:warn 3/5/2019, 9:50:51 PM No converter available for 'GL-B-007Z' with cid 'genBasic', type 'attReport' and data '{"cid":"genBasic","data":{"manufacturerName":"GLEDOPTO","modelId":"GL-B-007Z","powerSource":1}}'
  zigbee2mqtt:warn 3/5/2019, 9:50:51 PM Please see: https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html.
  zigbee2mqtt:warn 3/5/2019, 9:50:51 PM No converter available for 'GL-B-007Z' with cid 'genBasic', type 'devChange' and data '{"cid":"genBasic","data":{"manufacturerName":"GLEDOPTO","modelId":"GL-B-007Z","powerSource":1}}'
  zigbee2mqtt:warn 3/5/2019, 9:50:51 PM Please see: https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html.
```

**Comment by alinelena** [comment]
@milakov  here the same just more of them. I do not know how is your network but mine with the latest dev and latest firmware from dev seems to be centred around the coordinator not the usual mesh. somehow I think the coordinator gets overwhelmed by requests to join. apparently i have 29 devices joined. also all the warning seem to be on powered devices... ikea lights or xiaomi socket.

**Comment by milakov** [comment]
@alinelena I have 24 devices and I am pretty sure they are connected via routers (5 of them).

**Comment by Koenkk** [comment]
> I am getting this with the very latest dev both for GL-B-007Z and GL-B-008Z:
> 
> ```
>   zigbee2mqtt:warn 3/5/2019, 9:50:51 PM No converter available for 'GL-B-007Z' with cid 'genBasic', type 'attReport' and data '{"cid":"genBasic","data":{"manufacturerName":"GLEDOPTO","modelId":"GL-B-007Z","powerSource":1}}'
>   zigbee2mqtt:warn 3/5/2019, 9:50:51 PM Please see: https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html.
>   zigbee2mqtt:warn 3/5/2019, 9:50:51 PM No converter available for 'GL-B-007Z' with cid 'genBasic', type 'devChange' and data '{"cid":"genBasic","data":{"manufacturerName":"GLEDOPTO","modelId":"GL-B-007Z","powerSource":1}}'
>   zigbee2mqtt:warn 3/5/2019, 9:50:51 PM Please see: https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html.
> ```

Should be fixed in latest dev

**Comment by milakov** [comment]
@Koenkk Now I am seeing this:
```  zigbee2mqtt:warn 3/8/2019, 11:19:33 AM No converter available for 'GL-B-007Z' with cid 'genBasic', type 'devChange' and data '{"cid":"genBasic","data":{"manufacturerName":"GLEDOPTO","modelId":"GL-B-007Z","powerSource":1}}'
  zigbee2mqtt:warn 3/8/2019, 11:19:33 AM Please see: https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html.
  zigbee2mqtt:info 3/8/2019, 11:19:46 AM MQTT publish: topic 'zigbee2mqtt/0x00158d00024f2e59', payload '{"state":"OFF","power":0,"voltage":204,"consumption":103.39,"temperature":32,"linkquality":34,"last_seen":"2019-03-08T11:19:46+03:00"}'
  zigbee2mqtt:info 3/8/2019, 11:20:03 AM Device incoming...
  zigbee2mqtt:info 3/8/2019, 11:20:03 AM MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"type":"pairing","message":"device incoming"}'
  zigbee2mqtt:warn 3/8/2019, 11:20:04 AM No converter available for 'GL-B-007Z' with cid 'genBasic', type 'devChange' and data '{"cid":"genBasic","data":{"manufacturerName":"GLEDOPTO","modelId":"GL-B-007Z","powerSource":1}}'
  zigbee2mqtt:warn 3/8/2019, 11:20:04 AM Please see: https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html.
  zigbee2mqtt:info 3/8/2019, 11:20:32 AM MQTT publish: topic 'zigbee2mqtt/0x00158d0002561266/availability', payload 'online'
  zigbee2mqtt:info 3/8/2019, 11:20:33 AM MQTT publish: topic 'zigbee2mqtt/0x00124b001b505747/availability', payload 'online'
  zigbee2mqtt:info 3/8/2019, 11:20:34 AM Device incoming...
  zigbee2mqtt:info 3/8/2019, 11:20:34 AM MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"type":"pairing","message":"device incoming"}'
  zigbee2mqtt:warn 3/8/2019, 11:20:35 AM No converter available for 'GL-B-008Z' with cid 'genBasic', type 'devChange' and data '{"cid":"genBasic","data":{"manufacturerName":"GLEDOPTO","modelId":"GL-B-008Z","powerSource":1}}'
  zigbee2mqtt:warn 3/8/2019, 11:20:35 AM Please see: https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html.
  zigbee2mqtt:info 3/8/2019, 11:20:46 AM MQTT publish: topic 'zigbee2mqtt/0x00158d00024f2e59/availability', payload 'online'```

**Comment by Koenkk** [comment]
@milakov thanks, will be ignored in the next version.

**Comment by apeeters** [comment]
After enabling reporting on firmware 20190223, latest hassio zigbee2mqtt-edge I get the following error a few seconds after start. It seems to kill the cc2531: I have to unplug & plug it to get it working again.

`zigbee2mqtt:error 3/10/2019, 10:23:33 PM 
	{"message":"request timeout","stack":"Error: request timeout\n    at CcZnp.<anonymous> (/app/node_modules/cc-znp/lib/ccznp.js:261:22)\n    at Object.onceWrapper (events.js:273:13)\n    at CcZnp.emit (events.js:182:13)\n    at Timeout.<anonymous> (/app/node_modules/cc-znp/lib/ccznp.js:240:18)\n    at ontimeout (timers.js:436:11)\n    at tryOnTimeout (timers.js:300:5)\n    at listOnTimeout (timers.js:263:5)\n    at Timer.processTimers (timers.js:223:10)"}`

**Comment by buffeh** [comment]
**apeeters** - I am running the same firmware and same latest hassio zigbee2mqtt-edge and I seem to get the same error. 

I enabled reporting, it appeared to work fine (briefly), then the stick stopped responding and restarts resulted in the above error. I actually had to reflash the stick with the latest firmware to get zigbee-shepherd to start working again, multiple restarts didn't help until I did this. I then was able to recreate the issue by enabling reporting again and for me this reliably "breaks the stick" requiring a reflash to get things up and running again.

Everything seems to work "ok" if I don't enable reporting, although even with latest firmware and edge version I still have stability issues if I turn off / on all my lights at once, causing zigbee2mqtt to crash and usually requiring me to unplug and replug the stick to get it to start up successfully again.

My network is about 15 Ikea Tradfri bulbs and a bunch of Xiaomi wireless 1 button and 2 button switches.

**Comment by Koenkk** [comment]
@apeeters do you experience the same with the 20190109 firmware? https://github.com/Koenkk/Z-Stack-firmware/tree/9215ec3183b6febe4bac3ebb5240f4285c69e6f0/coordinator/CC2531/bin

**Comment by apeeters** [comment]
@Koenkk I'd rather not downgrade below 20190215 to avoid having to re-pair everything. I will downgrade to 20190218 and try again.

**Comment by apeeters** [comment]
@Koenkk It seems I completely killed it: I can hardly turn on/of 1 group before I get timeouts and have to unplug the device to resurrect it. After swapping in another cc2531, everything works fine again (20190218 on 1.2.1 on hassio).

**Comment by apeeters** [comment]
I spoke too soon: the new one also crashes after a while (a few minutes for 20190218 to 15 minutes for 20190223). I'll keep a close eye on it.

**Comment by buffeh** [comment]
@apeeters it sounds like you have the same issue as me, disabling reporting and reflashing the stick with latest firmware should get it up and running again, as I said before it seems enabling reporting for us with latest Edge and latest firmware kills the stick requiring a reflash!

**Comment by Koenkk** [comment]
@apeeters @buffeh please provide the complete log up-to the crash.

**Comment by apeeters** [comment]
FWIW, I also have failing pings to "ZNCZ02LM - Xiaomi Mi power plug ZigBee (Router)". I'll check what happens when I remove it.

**Comment by apeeters** [comment]
@Koenkk You can find the log here: https://hastebin.com/ovofupojev.apache
At 10:03:32 PM I turned group 8 on via mqtt, this triggers a change color event, but that does not matter since it was already crashed.

**Comment by apeeters** [comment]
> FWIW, I also have failing pings to "ZNCZ02LM - Xiaomi Mi power plug ZigBee (Router)". I'll check what happens when I remove it.

Does not have any impact.

**Comment by Koenkk** [comment]
@apeeters could you provide the debug log? (`log_level: debug`) http://www.zigbee2mqtt.io/configuration/configuration.html

**Comment by apeeters** [comment]
@Koenkk zigbee2mqtt hassio (non-edge) hangs almost immediately when enabling debug: https://hastebin.com/uquxilixix.makefile I'll let it run for a while to see what happens.

**Comment by Koenkk** [comment]
@apeeters this is really strange, it hangs before even starting, are you sure this is fixed by disabling reporting?

**Comment by apeeters** [comment]
@Koenkk This has nothing to do with reporting afaik, I used the same config (except for debug) for yesterday's and today's log. I currently see the following every 5 minutes:
`3/13/2019, 7:51:00 PM - debug: Saving state to file /share/zigbee2mqtt/state.json
`

**Comment by apeeters** [comment]
Wild guess: we are triggering a bug in winston logger? Seems there are quite a few bugs related to logging stopping.

**Comment by Koenkk** [comment]
@apeeters so when running with `log_level: info` again it starts?

**Comment by apeeters** [comment]
@Koenkk sometimes it does, sometimes it doesn't.  Currently can't explain why.

**Comment by apeeters** [comment]
dmesg log: https://hastebin.com/ayobiqezup.sql

**Comment by Koenkk** [comment]
@apeeters could you try re-flashing the stick?

**Comment by apeeters** [comment]
@Koenkk I already did multiple times. My issue is exactly this: https://github.com/Koenkk/Z-Stack-firmware/issues/46, maybe we move the discussion over there? Next thing I will try: power off a major part of the Zigbee network before starting zigbee2mqtt.

**Comment by Koenkk** [comment]
Let's move it there.

**Comment by apeeters** [comment]
I just setup reporting. It seems to work, but I get errors like the following for a few bulbs. Maybe delay/spread reporting messages in a similar way to regular control messages?

`3/18/2019, 8:30:58 PM - error: Failed to bind for reporting for 0x90fd9ffffe4d1be0 - lightingColorCtrl - 3 - (Error: Timed out after 10000 ms)`

**Comment by joseska** [comment]
Same error as @apeeters with "ZNCZ02LM - Xiaomi Mi power plug ZigBee (Router)"

**Comment by Koenkk** [comment]
@apeeters @joseska can you try by changing https://github.com/Koenkk/zigbee2mqtt/blob/master/lib/util/zigbeeQueue.js#L1 to `const maxSimultaneouslyRunning = 3;`

**Comment by glentakahashi** [comment]
I've noticed it tries to bind reporting to devices that don't need reporting, for example my xiaomi aqara door/window sensors. It's trying to bind them to genOnOff, which almost always fails because they're battery powered and always sleeping.

**Comment by Koenkk** [comment]
@glentakahashi good catch, should be fixed in the latest dev branch.

**Comment by lassieee** [comment]
Hi,
I have three zigbee2mqtt groups, each with their own Ikea E1524 button and one or more Ikea bulbs (mix of e27, e14 and gu10). I have enabled reporting so these bulbs show up as ON when the E1524 turns the bulb on, but when I have reporting on, zigbee2mqtt stops working within a day and the only way I can get zigbee2mqtt to start again is to reflash the CC2531. I've been having this for a few weeks now.
I have two CC2531 sticks and two raspberry pi 3's, and if I switch hardware the issue is the same. I'm running the latest version for both the CC2531 firmware and the zigbee2mqtt docker container.
The error zigbee2mqtt gives is:
```
Using` '/app/data' as data directory

> zigbee2mqtt@1.3.1 start /app
> node index.js

  zigbee2mqtt:info 4/18/2019, 6:25:25 AM Logging to directory: '/app/data/log/2019-04-18.06-25-25'
  zigbee2mqtt:info 4/18/2019, 6:25:27 AM Starting zigbee2mqtt version 1.3.1 (commit #unknown)
  zigbee2mqtt:info 4/18/2019, 6:25:27 AM Starting zigbee-shepherd
  zigbee2mqtt:info 4/18/2019, 6:25:33 AM Error while starting zigbee-shepherd, attempting to fix... (takes 60 seconds)
  zigbee2mqtt:info 4/18/2019, 6:26:33 AM Starting zigbee-shepherd
  zigbee2mqtt:error 4/18/2019, 6:26:40 AM Error while starting zigbee-shepherd!
  zigbee2mqtt:error 4/18/2019, 6:26:40 AM Press the reset button on the stick (the one closest to the USB) and start again
  zigbee2mqtt:error 4/18/2019, 6:26:40 AM Failed to start
        {"message":"request timeout","stack":"Error: request timeout\n    at CcZnp.<anonymous> (/app/node_modules/cc-znp/lib/ccznp.js:261:22)\n    at Object.onceWrapper (events.js:273:13)\n    at CcZnp.emit (events.js:182:13)\n    at Timeout.<anonymous> (/app/node_modules/cc-znp/lib/ccznp.js:240:18)\n    at ontimeout (timers.js:436:11)\n    at tryOnTimeout (timers.js:300:5)\n    at listOnTimeout (timers.js:263:5)\n    at Timer.processTimers (timers.js:223:10)"}
  zigbee2mqtt:error 4/18/2019, 6:26:40 AM Exiting...
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! zigbee2mqtt@1.3.1 start: `node index.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the zigbee2mqtt@1.3.1 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     `/root/.npm/_logs/2019-04-18T06_27_15_943Z-debug.log
```

I disabled reporting this weekend and I had no issues, no crashed whatsoever. Yesterday I enabled reporting again and this morning it's broken again.
Is anyone else experiencing this, or have any pointers for me?
I'm running CC2531_DEFAULT_20190223.zip and zigbee2mqtt 1.3.1-arm32v6 (docker)

**Comment by Koenkk** [comment]
@lassieee can you try with the max stability firmware? https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator

**Comment by lassieee** [comment]
@Koenkk  thanks Koen, I'll try that and report back.

**Comment by lassieee** [comment]
> @lassieee can you try with the max stability firmware? https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator

It's been three days now, without issues, so the max stability firmware seems the way to go for me. Many thanks @Koenkk !

**Comment by sanderjonkers** [comment]
I'm testing with reporting ON, but I see some errors in the log.
I'm using the zigbee2mqtt add-on in hassio, V1.3.1

For a Ikea Tradfri GU10, the log contains:
`zigbee2mqtt:error 5/3/2019, 1:41:37 PM Failed to bind for reporting for 0xd0cf5efffedcbb6f - genOnOff - 1 - (Error: Timed out after 10000 ms)
zigbee2mqtt:error 5/3/2019, 1:42:27 PM Failed to bind for reporting for 0xd0cf5efffedcbb6f - genLevelCtrl - 1 - (Error: Timed out after 10000 ms)`

I also saw others reporting this earlier in this thread, but I couldn't find a fix.

**Comment by gdave321** [comment]
Seems reporting is NOT working correctly for a HUE lamp (Hue white ambiance E14 (8718696695203)).

I have an IKEA switch plus this hue light setup in a group 'testgrp'  sending ON to this group, there is a report from the IKEA it turned on but nothing from the Hue light.
Reportng does appear to be setup successfully:
hue bulb:
5/5/2019, 10:20:46 AM - debug: Successfully setup reporting for 0x00178801029b5654 - genOnOff - 1
ikea switch:
5/5/2019, 10:21:03 AM - debug: Successfully setup reporting for 0x000d6ffffedda4b3 - genOnOff - 1

Here is the group on, both light and switch actually do turn on just fine but no report from Hue:

5/5/2019, 11:02:32 AM - debug: Received MQTT message on 'zigbee2mqtt/testgrp/set' with data '{"state": "ON"}'
5/5/2019, 11:02:32 AM - info: Zigbee publish to group '1', genOnOff - on - {} - {"manufSpec":0,"disDefaultRsp":0} - null
5/5/2019, 11:02:32 AM - info: MQTT publish: topic 'zigbee2mqtt/testgrp', payload '{"state":"ON","color":{"x":0.457,"y":0.41},"brightness":1}'
5/5/2019, 11:02:32 AM - debug: Received zigbee message of type 'attReport' with data '{"cid":"genOnOff","data":{"onOff":1}}' of device 'TRADFRI control outlet' (0x000d6ffffedda4b3) of endpoint 1
5/5/2019, 11:02:32 AM - info: MQTT publish: topic 'zigbee2mqtt/testgrp', payload '{"state":"ON","color":{"x":0.457,"y":0.41},"brightness":1}'

**Comment by gdave321** [comment]
BTW I since I noticed in the publish to the group the brightness was set to '1' I repeated the test by sending state ON brightness 255 to the group, however doesn't make a difference, IKEA reports it's on, the HUE doesn't report anything.

5/5/2019, 11:19:30 AM - debug: Received MQTT message on 'zigbee2mqtt/testgrp/set
' with data '{"state": "ON", "brightness": 255}'
5/5/2019, 11:19:30 AM - info: Zigbee publish to group '1', genLevelCtrl - moveTo
LevelWithOnOff - {"level":255,"transtime":0} - {"manufSpec":0,"disDefaultRsp":0}  - null
5/5/2019, 11:19:30 AM - info: MQTT publish: topic 'zigbee2mqtt/testgrp', payload
 '{"state":"ON","color":{"x":0.457,"y":0.41},"brightness":255}'
5/5/2019, 11:19:30 AM - debug: Received zigbee message of type 'attReport' with 
data '{"cid":"genOnOff","data":{"onOff":1}}' of device 'TRADFRI control outlet' 
(0x000d6ffffedda4b3) of endpoint 1
5/5/2019, 11:19:30 AM - info: MQTT publish: topic 'zigbee2mqtt/testgrp', payload '{"state":"ON","color":{"x":0.457,"y":0.41},"brightness":255}'

**Comment by Koenkk** [comment]
@gdave321 https://github.com/Koenkk/zigbee2mqtt/issues/1064#issuecomment-468962505

**Comment by gdave321** [comment]
@Koenkk
If you reference the remark that Philips Hue doesn't report brightness level, yes I've seen that. But just seeing on/off would be great already.

**Comment by Koenkk** [comment]
@gdave321 I see, but note that this feature completely depends on the bulb itself. If you still have a Hue hub you could try updating the software of the bulb.

**Comment by End0rphi** [comment]
Reporting is working correctly with Müller Tint lamps, changing state via Tint remote.
Payload of reported state is e.g.: "'{"state":"ON","linkquality":31,"brightness":1,"color_temp":250,"color_mode":2,"color":{"x":0.458,"y":0.41}}'"

Changing brightness or color temperature via the Tint remote also reports back to the controller.

**Comment by brettyj** [comment]
I'm having the same issue as @gdave321 - my hue bulbs are showing as having registered for reporting fine, but when changing via a zigbee group, the individual bulbs are not updating - and I'm seeing no reports coming through in debug mode.

I've only just switched my bulbs from Hue hub so they are all on the latest software versions.
Got the following bulbs:

  Philips Hue white ambiance GU10 (Router)
  Philips Hue white ambiance GU10 (Router)
  Philips Hue dimmer switch (EndDevice)
  Philips Hue white ambiance GU10 (Router)
  Philips Hue Lux A19 bulb E27 (Router)
  Philips Hue Lux A19 bulb E27 (Router)
  Philips Hue Lux A19 bulb E27 (Router)
  Philips Hue white and color ambiance LightStrip plus (Router)
  Philips Hue white and color ambiance E26/E27/E14 (Router)
  Philips Hue Lux A19 bulb E27 (Router)

Also noticed that they are all showing as offline in the network map but seem to be working fine.

**Comment by apeeters** [comment]
I have the impression that reporting is flooding my network with messages. Although reporting is disabled in the advanced settings, some Tradfri lights still report their status.

Is it possible to disable reporting again after enabling it?

**Comment by Koenkk** [comment]
For tradfri you have to powercycle them after disabling reporting.

**Comment by apeeters** [comment]
Reporting did flood my network. After disabling and powercycling the Tradfri devices my network is stable again.

**Comment by groenmarsmannetje** [comment]
Muller Tint light bulbs flood network with state reports without any actual state change. Observed in dev branch

**Comment by Koenkk** [comment]
@groenmarsmannetje can you provide the debug logging of this?

To enable debug logging set in `configuration.yaml`:
```yaml
advanced:
  log_level: debug
```

**Comment by gabe565** [comment]
Really loving Zigbee2mqtt, but I thought I'd chime in here. I can't get reporting working for any of my Hue bulbs on the latest release or on the dev version.

On the latest release, it looks like reporting gets setup fine, but then I never get any reports when I trigger a group.
On the dev version, reporting fails to be setup on launch, just giving a timeout error.

Hue bulbs are all on the latest firmware.

Example log entry on dev:
```
  zigbee2mqtt:error 9/23/2019, 9:50:30 AM Failed to setup reporting for '0x00178801027fb9d2' - Error: Timeout - 49867 - 11 - 4 - 7 after 10000ms
    at Timeout.object.timer.setTimeout [as _onTimeout] (/app/node_modules/zigbee-herdsman/dist/utils/waitress.js:44:24)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10)
```

**Comment by BudBundi** [comment]
I have some gledopto GU10 spots with due to a firmware bug, did report but doesn't answer the setup.
So if I enable the reporting in the dev container it slows down the network a lot due the fact zigbee2mqtt try it again and again.

`zigbee2mqtt:error 9/23/2019, 9:00:50 PM Failed to setup reporting for '0x00124b001d3aabf5' - Error: Timeout - 62245 - 11 - 59 - 7 after 10000ms
    at Timeout.object.timer.setTimeout [as _onTimeout] (/app/node_modules/zigbee-herdsman/dist/utils/waitress.js:44:24)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10)`

Is it possible to disable reporting on device level or can you make it possible?

**Comment by Koenkk** [comment]
@BudBundi could you explain more about the firmware bug?

**Comment by hobbyquaker** [comment]
> Muller Tint light bulbs flood network with state reports without any actual state change. Observed in dev branch

Can confirm that, they are sending attributeReports in 3sec intervals on the genLevelCtrl and lightingColorCtrl cluster.
The genOnOff seems not to have that problem.

**Comment by Koenkk** [comment]
@hobbyquaker I think this can also be fixed by changing:

- https://github.com/Koenkk/zigbee2mqtt/blob/dev/lib/extension/deviceReport.js#L8 to `minimumReportInterval: 3, maximumReportInterval: 300, reportableChange: 1,`
- https://github.com/Koenkk/zigbee2mqtt/blob/dev/lib/extension/deviceReport.js#L11 to `const reportKey = 1;`

**Comment by Kryzek** [comment]
While re-pairing my devices got this error with my **Hue white and color ambiance E26/E27/E14 (9290012573A)** -bulb:
```
error 2019-11-14T17:05:05: Failed to setup reporting for '0x001788010278b40a' - Error: Timeout - 4247 - 11 - 2 - 6 - 7 after 10000ms
    at Timeout.object.timer.setTimeout [as _onTimeout] (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/utils/waitress.js:44:24)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10)
```

This device entry from .db
```
{
    "id": 17,
    "type": "Router",
    "ieeeAddr": "0x0017880104a5c9f1",
    "nwkAddr": 15928,
    "manufId": 4107,
    "manufName": "Philips",
    "powerSource": "Mains (single phase)",
    "modelId": "LWB010",
    "epList": [
        11,
        242
    ],
    "endpoints": {
        "11": {
            "profId": 49246,
            "epId": 11,
            "devId": 256,
            "inClusterList": [
                0,
                3,
                4,
                5,
                6,
                8,
                4096
            ],
            "outClusterList": [
                25
            ],
            "clusters": {
                "genBasic": {
                    "attributes": {
                        "modelId": "LWB010",
                        "manufacturerName": "Philips",
                        "powerSource": 1,
                        "zclVersion": 1,
                        "appVersion": 2,
                        "stackVersion": 1,
                        "hwVersion": 1,
                        "dateCode": "20170908",
                        "swBuildId": "1.29.0_r21169"
                    }
                },
                "genOnOff": {
                    "attributes": {
                        "onOff": 1
                    }
                }
            },
            "binds": [
                {
                    "cluster": 6,
                    "type": "endpoint",
                    "deviceIeeeAddress": "0x00124b001ca16964",
                    "endpointID": 1
                },
                {
                    "cluster": 8,
                    "type": "endpoint",
                    "deviceIeeeAddress": "0x00124b001ca16964",
                    "endpointID": 1
                }
            ]
        },
        "242": {
            "profId": 41440,
            "epId": 242,
            "devId": 97,
            "inClusterList": [
                33
            ],
            "outClusterList": [
                33
            ],
            "clusters": {},
            "binds": []
        }
    },
    "appVersion": 2,
    "stackVersion": 1,
    "hwVersion": 1,
    "dateCode": "20170908",
    "swBuildId": "1.29.0_r21169",
    "zclVersion": 1,
    "interviewCompleted": true,
    "meta": {
        "reporting": 1
    }
}
```

**Comment by Kryzek** [comment]
When I restarted my Zigbee2MQTT I noticed that some Hue  bulbs spit out an error.
```
Hue White Single bulb B22 (8718696449691)
zigbee2mqtt:error 2019-11-21 13:43:05: Failed to setup reporting for '0x0017880102b3b0ab' - Error: Timeout - 33592 - 11 - 2 - 6 - 7 after 10000ms
    at Timeout.object.timer.setTimeout [as _onTimeout] (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/utils/waitress.js:44:24)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10)
	
Hue white and color ambiance E26/E27/E14 (9290012573A)
zigbee2mqtt:error 2019-11-21 13:43:05: Failed to setup reporting for '0x001788010278b40a' - Error: Timeout - 26093 - 11 - 3 - 6 - 7 after 10000ms
    at Timeout.object.timer.setTimeout [as _onTimeout] (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/utils/waitress.js:44:24)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10)
```

**Comment by bodiroga** [comment]
+1 @Kryzek!

In my case I have 3x Philips Hue bulbs (quite old): 2x Philips Hue Lux A19 bulb E27 and 1x Philips Hue white A60 bulb E27.

And here you can find the log messages:

```
zigbee2mqtt:error 2019-12-13T22:45:31: Failed to setup reporting for '0x001788011040a46b' - Error: Timeout - 17728 - 11 - 2 - 6 - 7 after 10000ms
    at Timeout.object.timer.setTimeout [as _onTimeout] (/app/node_modules/zigbee-herdsman/dist/utils/waitress.js:44:24)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10)
zigbee2mqtt:error 2019-12-13T22:45:31: Failed to setup reporting for '0x0017880100db65c0' - Error: Timeout - 49932 - 11 - 3 - 6 - 7 after 10000ms
    at Timeout.object.timer.setTimeout [as _onTimeout] (/app/node_modules/zigbee-herdsman/dist/utils/waitress.js:44:24)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10)
zigbee2mqtt:error 2019-12-13T22:45:41: Failed to setup reporting for '0x0017880100db6159' - Error: Timeout - 46369 - 11 - 4 - 6 - 7 after 10000ms
    at Timeout.object.timer.setTimeout [as _onTimeout] (/app/node_modules/zigbee-herdsman/dist/utils/waitress.js:44:24)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10)
```

My coordinator is a CC2652R and I'm using the latest stable version of program: "zigbee2mqtt version 1.7.1" (commit #b459c35).

What type of log reports would you need, @Koenkk?

Thank you very much and best regards,

Aitor

**Comment by czikus** [comment]
I can confirm the problem with all my hue bulbs on the latest dev (1.7.1+dev (commit #1972eb4)):

```
error 2019-12-14 11:00:27: Failed to setup reporting for '0x0017880110450438' - Error: Timeout - 32594 - 11 - 2 - 6 - 7 after 10000ms
    at Timeout._onTimeout (/home/ubuntu/.dot/modules/40_dot-module-pronobis-machine-iot2/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/utils/waitress.js:44:24)
    at listOnTimeout (internal/timers.js:536:17)
    at processTimers (internal/timers.js:480:7)
```

**Comment by Koenkk** [comment]
@bodiroga  Sniffing the traffic is needed to pinpoint the problem (https://www.zigbee2mqtt.io/how_tos/how_to_sniff_zigbee_traffic.html).

**Comment by bodiroga** [comment]
Hi @Koenkk!

I have followed the steps described in your link (as always, amazing documentation, well done sir) and attached to this message you can find the packages captured with wireshark. This is the order I have followed:

- Start wireshark with ```sudo whsniff -c 11 | wireshark -k -i -```
- Restart the docker container with ```docker restart zigbee2mqtt```
- Wait until the timeout messages appear in the log file (```docker logs -f zigbee2mqtt```)

As in the previous message, the error messages have been:

```
zigbee2mqtt:error 2019-12-15T20:44:29: Failed to setup reporting for '0x001788011040a46b' - Error: Timeout - 17728 - 11 - 2 - 6 - 7 after 10000ms
    at Timeout.object.timer.setTimeout [as _onTimeout] (/app/node_modules/zigbee-herdsman/dist/utils/waitress.js:44:24)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10)
zigbee2mqtt:error 2019-12-15T20:44:29: Failed to setup reporting for '0x0017880100db65c0' - Error: Timeout - 49932 - 11 - 3 - 6 - 7 after 10000ms
    at Timeout.object.timer.setTimeout [as _onTimeout] (/app/node_modules/zigbee-herdsman/dist/utils/waitress.js:44:24)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10)
zigbee2mqtt:error 2019-12-15T20:44:39: Failed to setup reporting for '0x0017880100db6159' - Error: Timeout - 46369 - 11 - 4 - 6 - 7 after 10000ms
    at Timeout.object.timer.setTimeout [as _onTimeout] (/app/node_modules/zigbee-herdsman/dist/utils/waitress.js:44:24)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10)
```
[zigbee2mqtt.zip](https://github.com/Koenkk/zigbee2mqtt/files/3965432/zigbee2mqtt.zip)

I don't understand what the packages captured by wireshark mean, but I hope that they can be useful. If you need anything else, don't hesitate to ask, please :wink: 

Best regards,

Aitor

**Comment by Koenkk** [comment]
@bodiroga it seems that your bulbs don't support reporting (probably because of the old firmware???). This can be seen in the log:

![image](https://user-images.githubusercontent.com/2892853/70937763-30190600-2045-11ea-9533-13c88b560acf.png)

**Comment by bodiroga** [comment]
> @bodiroga it seems that your bulbs don't support reporting (probably because of the old firmware???). 

Oh, that's a shame :cry:  It's true that the bulbs are quite old, but I though that they could be able to report their status, as they have received the newest firmware not so long ago (20181121): 

```json
{"ieeeAddr":"0x0017880100db65c0","type":"Router","networkAddress":xxxxx,"model":"433714","friendly_name":"philips-white-1","manufacturerID":4107,"manufacturerName":"Philips","powerSource":"Mains (single phase)","modelID":"LWB004","hardwareVersion":1,"softwareBuildID":"5.127.1.26420","dateCode":"20181121","lastSeen":1576599446324}
```
Before pairing the bulbs with the CC2652R, I checked if a new firmware was available for them, but the one from 11/2018 was the latest one. I did it using the first generation Philips Hue Bridge, but I don't know if that matters, as far as I know, it shouldn't.

Anyway, many thanks @Koenkk for having a look at my log file :wink:

**Comment by Koenkk** [comment]
Yes please open a bug at herdsman, can you also provide the herdsman debug logging in this report when setting up the reporting fails?

To enable herdsman debug logging, see https://www.zigbee2mqtt.io/information/debug.html#zigbee-herdsman-debug-logging

**Comment by Odianosen25** [comment]
Hello @Koenkk,

Thanks again for this great software. I am using this feature and I have the following setup
- Have 2 Philips Hue Bulbs in a Group
- Have the Philips Hue Dimmer which I binded to the Philips Hue Bulbs Group
- When I turn on/off the Group using the dimmer, no report is send back from the group, neither the bulbs
- If I increase the brightness with it, the bulbs report their brightness level individually, but nothing on the group
- Also though the brightness is reported by the bulbs individually (not the group), the `state` reports as `OFF`, even though the bulbs are `ON`; of course there is no brightness without being ON anyway

I am on 1.10 .0 and using the CC2531. 

Regards

**Comment by chaliluc** [comment]
After installing the latest dev branch, I activated reporting in my configuration.yaml for 3 Fyrtur roller blinds but it only works for two of them. 

Below is an extract of my configuration.yaml file:
`advanced:
  report: true
devices:
  '0xd0cf5efffeea6130':
    friendly_name: salon_store_1
    retrieve_state: true
  '0xd0cf5efffeef59d3':
    friendly_name: salon_store_2
    retrieve_state: true
  '0xd0cf5efffed98efe':
    friendly_name: salon_store_3
    retrieve_state: true
`

The database.db file for those 3 devices looks like this:
`{
   "id":6,
   "type":"EndDevice",
   "ieeeAddr":"0xd0cf5efffeea6130",
   "nwkAddr":37388,
   "manufId":4476,
   "manufName":"IKEA of Sweden",
   "powerSource":"Battery",
   "modelId":"FYRTUR block-out roller blind",
   "epList":[
      1
   ],
   "endpoints":{
      "1":{
         "profId":260,
         "epId":1,
         "devId":514,
         "inClusterList":[
            0,
            1,
            3,
            4,
            5,
            32,
            258,
            4096,
            64636
         ],
         "outClusterList":[
            25,
            4096
         ],
         "clusters":{
            "genBasic":{
               "attributes":{
                  "modelId":"FYRTUR block-out roller blind",
                  "manufacturerName":"IKEA of Sweden",
                  "powerSource":3,
                  "zclVersion":3,
                  "appVersion":34,
                  "stackVersion":99,
                  "hwVersion":1,
                  "dateCode":"20190311",
                  "swBuildId":"2.2.007"
               }
            },
            "closuresWindowCovering":{
               "attributes":{
                  "currentPositionLiftPercentage":0
               }
            },
            "genPowerCfg":{
               "attributes":{
                  "batteryPercentageRemaining":85
               }
            }
         },
         "binds":[
            {
               "cluster":1,
               "type":"endpoint",
               "deviceIeeeAddress":"0x00124b00194306a5",
               "endpointID":1
            },
            {
               "cluster":258,
               "type":"endpoint",
               "deviceIeeeAddress":"0x00124b00194306a5",
               "endpointID":1
            }
         ]
      }
   },
   "appVersion":34,
   "stackVersion":99,
   "hwVersion":1,
   "dateCode":"20190311",
   "swBuildId":"2.2.007",
   "zclVersion":3,
   "interviewCompleted":true,
   "meta":{
      "configured":2,
      "reporting":1
   },
   "lastSeen":1583860545569
}
{
   "id":7,
   "type":"EndDevice",
   "ieeeAddr":"0xd0cf5efffeef59d3",
   "nwkAddr":32168,
   "manufId":4476,
   "manufName":"IKEA of Sweden",
   "powerSource":"Battery",
   "modelId":"FYRTUR block-out roller blind",
   "epList":[
      1
   ],
   "endpoints":{
      "1":{
         "profId":260,
         "epId":1,
         "devId":514,
         "inClusterList":[
            0,
            1,
            3,
            4,
            5,
            32,
            258,
            4096,
            64636
         ],
         "outClusterList":[
            25,
            4096
         ],
         "clusters":{
            "genBasic":{
               "attributes":{
                  "modelId":"FYRTUR block-out roller blind",
                  "manufacturerName":"IKEA of Sweden",
                  "powerSource":3,
                  "zclVersion":3,
                  "appVersion":34,
                  "stackVersion":99,
                  "hwVersion":1,
                  "dateCode":"20190311",
                  "swBuildId":"2.2.007"
               }
            },
            "closuresWindowCovering":{
               "attributes":{
                  "currentPositionLiftPercentage":19
               }
            },
            "genPowerCfg":{
               "attributes":{
                  "batteryPercentageRemaining":100
               }
            }
         },
         "binds":[
            {
               "cluster":1,
               "type":"endpoint",
               "deviceIeeeAddress":"0x00124b00194306a5",
               "endpointID":1
            },
            {
               "cluster":258,
               "type":"endpoint",
               "deviceIeeeAddress":"0x00124b00194306a5",
               "endpointID":1
            }
         ]
      }
   },
   "appVersion":34,
   "stackVersion":99,
   "hwVersion":1,
   "dateCode":"20190311",
   "swBuildId":"2.2.007",
   "zclVersion":3,
   "interviewCompleted":true,
   "meta":{
      "configured":2,
      "reporting":1
   },
   "lastSeen":1583860591537
}
{
   "id":8,
   "type":"EndDevice",
   "ieeeAddr":"0xd0cf5efffed98efe",
   "nwkAddr":51147,
   "manufId":4476,
   "manufName":"IKEA of Sweden",
   "powerSource":"Battery",
   "modelId":"FYRTUR block-out roller blind",
   "epList":[
      1
   ],
   "endpoints":{
      "1":{
         "profId":260,
         "epId":1,
         "devId":514,
         "inClusterList":[
            0,
            1,
            3,
            4,
            5,
            32,
            258,
            4096,
            64636
         ],
         "outClusterList":[
            25,
            4096
         ],
         "clusters":{
            "genBasic":{
               "attributes":{
                  "modelId":"FYRTUR block-out roller blind",
                  "manufacturerName":"IKEA of Sweden",
                  "powerSource":3,
                  "zclVersion":3,
                  "appVersion":34,
                  "stackVersion":99,
                  "hwVersion":1,
                  "dateCode":"20190311",
                  "swBuildId":"2.2.009"
               }
            },
            "closuresWindowCovering":{
               "attributes":{
                  "currentPositionLiftPercentage":17
               }
            }
         },
         "binds":[
            {
               "cluster":1,
               "type":"endpoint",
               "deviceIeeeAddress":"0x00124b00194306a5",
               "endpointID":1
            },
            {
               "cluster":258,
               "type":"endpoint",
               "deviceIeeeAddress":"0x00124b00194306a5",
               "endpointID":1
            }
         ]
      }
   },
   "appVersion":34,
   "stackVersion":99,
   "hwVersion":1,
   "dateCode":"20190311",
   "swBuildId":"2.2.009",
   "zclVersion":3,
   "interviewCompleted":true,
   "meta":{
      "configured":2,
      "reporting":1
   },
   "lastSeen":1583860579950
} 
`

The first one (id = 6) is the one that doesn't work, while the two others (id = 7 and id=8) are reporting their positions just fine. 

Another very strange thing is that those 3 blinds are the exact same, purchased at the same time, and run the latest firmware. However, the third one doesn't report on its battery, and has the complete attribute "genPowerCfg" missing in the database.db.

Is there something to do to fix this?

**Comment by shirou93** [comment]
Report feature work with xiaomi bulbs? This vendor not in table.

**Comment by magnushacker** [comment]
On https://www.zigbee2mqtt.io/information/report.html#support, there is a footnote that says "The color/brightness of a Trådfri bulb can be changed while the state=off, it also reports back the change.". This doesn't work for me.

I'd like to use this to set the brightness of my bedside lamp to 10 during the night, so that it's not super bright the first time I turn it on in the morning. The remote is bound to a group with the bulb in it, so I can control the bulb directly without z2m. Using the remote to turn on however always sets the same brightness as when I turned it off with the remote.

**Comment by michael-sabrnak-swi** [comment]
I experience same issue. I use Ikea toggle switch with bulb paired into the same group. If I control light via HA the status of light is updated.

However toggle switch button will report an action to the z2m but the bulb will not report status on/off back to z2m.

**Comment by MyKEms** [comment]
I experience same issue. I use Ikea toggle switch with bulb paired into the same group. If I control light via HA the status of light is updated.

However toggle switch button will report an action to the z2m but the bulb will not report status on/off back to z2m.

**Comment by jd1900** [comment]
It seems that new Gledopto LED controllers (Zigbee LED controller RGB + CCT pro model (GL-C-008P)) support reporting. At least for state and brightness.

**Comment by elgerg** [comment]
@jd1900 How did you set up and configure this for the GL-C-008P?

I have one and when the device is altered using a remote there is no report of the change.

This is the remote for the led controller:
https://www.aliexpress.com/item/1005002797042241.html?spm=a2g0s.9042311.0.0.12d84c4dSCBzTS

**Comment by elgerg** [comment]
Ok, i have managed to figure out what needs setting to get the on/off state:
![image](https://user-images.githubusercontent.com/24628571/124469387-36d7ad00-dd92-11eb-81f1-3d79a8e96d12.png)

Does anyone know how to get it to report the change in brightness/colour?

Thanks
Alex
